deobfuscator/src/test/java/net/runelite/deob/updater/AnnotationCleaner.java

```diff
@@ -83,4 +86,135 @@ public void checkMappings() throws Exception
 			}
 		}
 	}
+
+	@Test
+	@Ignore
+	public void fixMappings() throws Exception
+	{
+		File client = new File(properties.getRsClient());
+		ClassGroup group = JarUtil.loadJar(client);
+		int impl = 0,
+			meth = 0,
+			field = 0;
+
+		for (ClassFile c : group.getClasses())
+		{
+			if (c.getName().contains("runelite"))
+			{
+				continue;
+			}
+
+			log.debug("Checking {}", c.toString());
+
+			String implementingName = DeobAnnotations.getImplements(c);
+			if (!Strings.isNullOrEmpty(implementingName))
+			{
+				// Still error here cause I don't wanna call classes dumb shit
+				assertEquals(c + " implements " + implementingName + " but is called " + c.getClassName(), implementingName, c.getClassName());
+			}
+			else
+			{
+				if (!Deob.isObfuscated(c.getClassName()))
+				{
+					Annotations an = c.getAnnotations();
+
+					Annotation implAn = new Annotation(an);
+					implAn.setType(DeobAnnotations.IMPLEMENTS);
+
+					Element value = new Element(implAn);
+					value.setValue(c.getClassName());
+					value.setName("value");
+
+					implAn.addElement(value);
+					an.addAnnotation(implAn);
+					impl++;
+				}
+			}
+
+			for (Field f : c.getFields())
+			{
+				Annotations an = f.getAnnotations();
+
+				String fieldName = f.getName();
+				String exportedName = DeobAnnotations.getExportedName(an);
+
+				if (exportedName == null && Deob.isObfuscated(fieldName) ||  fieldName.equals(DeobAnnotations.getObfuscatedName(an)) || DeobAnnotations.getObfuscatedName(an) == null)
+				{
+					//Annotation exp = new Annotation(an);
```

Comment:
Remove commented code

Reply:
This code was really just born out of my laziness, (also copied from above)